samza-core/src/main/java/org/apache/samza/util/ConfigUtil.java

```diff
@@ -67,4 +72,42 @@ public static Config applyRewriter(Config config, String rewriterName) {
     LOG.info("Re-writing config with {}", rewriter);
     return rewriter.rewrite(rewriterName, config);
   }
+
+  /**
+   * Load full job config with {@link ConfigLoaderFactory} when present.
+   *
+   * @param original config
+   * @return full job config
+   */
+  public static Config loadConfig(Config original) {
+    JobConfig jobConfig = new JobConfig(original);
+    Config fullConfig = original;
+
+    if (jobConfig.getConfigLoaderFactory().isPresent()) {
+      ConfigLoaderFactory factory = ReflectionUtil.getObj(jobConfig.getConfigLoaderFactory().get(), ConfigLoaderFactory.class);
+      ConfigLoader loader = factory.getLoader(original.subset(ConfigLoaderFactory.CONFIG_LOADER_PROPERTIES_PREFIX));
+      // overrides config loaded with original config, which may contain overridden values.
+      fullConfig = ConfigUtil.rewriteConfig(override(loader.getConfig(), original));
+    }
+
+    return fullConfig;
+  }
+
+  /**
+   * Overrides original config with overridden values.
+   *
+   * @param original config to be overridden.
+   * @param overrides overridden values.
+   * @return the overridden config.
+   */
+  @SafeVarargs
+  public static Config override(Config original, Map<String, String>... overrides) {
```

Comment:
If this is public, can you please add a unit test for it? Or you could make it private.
You could also simplify it to only take in one extra argument since that is currently the only way it is used.

Reply:
Updated to private